plugins/ca/root-ca/src/main/java/org/apache/cloudstack/ca/provider/RootCAProvider.java

```diff
@@ -263,16 +263,20 @@ public SSLEngine createSSLEngine(final SSLContext sslContext, final String remot
         kmf.init(ks, getKeyStorePassphrase());
         tmf.init(ks);
 
-        final boolean authStrictness = rootCAAuthStrictness.value();
         final boolean allowExpiredCertificate = rootCAAllowExpiredCert.value();
 
-        TrustManager[] tms = new TrustManager[]{new RootCACustomTrustManager(remoteAddress, authStrictness, allowExpiredCertificate, certMap, caCertificate, crlDao)};
+        TrustManager[] tms = new TrustManager[]{createRootCACustomTrustManager(remoteAddress, allowExpiredCertificate, certMap, caCertificate, crlDao)};
+
         sslContext.init(kmf.getKeyManagers(), tms, new SecureRandom());
         final SSLEngine sslEngine = sslContext.createSSLEngine();
-        sslEngine.setNeedClientAuth(authStrictness);
+        sslEngine.setNeedClientAuth(rootCAAuthStrictness.value());
```

Comment:
@Slair1 any particular reason to use `rootCAAuthStrictness.value()` here and while creating RootCACustomTrustManager? there are some chances for the value here can be different from the one in RootCACustomTrustManager object created, as the config "ca.plugin.root.auth.strictness" is a dynamic setting. So, may be better to use it in earlier way (keeping in variable and use that) for the config value to be consistent.

Reply:
@sureshanaparti Yea, we can switch back to using the variable, thanks.  Only switched away from it since it was the only reference to that variable.